opentech/apply/funds/views.py

```diff
@@ -746,3 +746,20 @@ class RoundListView(SingleTableMixin, FilterView):
 
     def get_queryset(self):
         return RoundsAndLabs.objects.with_progress()
+
+
+@method_decorator(permission_required('funds.delete_applicationsubmission', raise_exception=True), name='dispatch')
+class SubmissionDeleteView(DeleteView):
+    model = ApplicationSubmission
+    success_url = reverse_lazy('funds:submissions:list')
+
+    def delete(self, request, *args, **kwargs):
+        submission = self.get_object()
+        messenger(
+            MESSAGES.DELETE_SUBMISSION,
+            user=request.user,
+            request=request,
+            submission=submission,
+        )
+        response = super().delete(request, *args, **kwargs)
```

Comment:
Ideally, the message should be sent after deletion of submission. But the messaging system requires Event model object which has a non-null foreign key to submission. We have to make some changes in the messaging system to make [event](https://github.com/OpenTechFund/opentech.fund/blob/a281060d595cc5e20c445a214689008814aef504/opentech/apply/activity/messaging.py#L710) optional. Suggestions?

Reply:
yeah, it should not be an issue.